K8s code executor

questcollector opened the pull request:
<!-- Thank you for your contribution! Please review https://microsoft.github.io/autogen/docs/Contribute before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

Using docker container for code executor is inconvenient when utilizing autogen through a kubernetes cluster.
Code executor using Kubernetes pod can address security concerns by leveraging Kubernetes RBAC policies.

New feature
- autogen.coding.kubernetes: module
- autogen.coding.kubernetes.PodCommandLineCodeExecutor: class
- extra requires [kubernetes] for setuptools

Test cases
- test/coding/test_kubernetes_commandline_code_executor.py: test case
- test/coding/test_kubernetes_commandline_code_executor.md: test case guide

Documentation
- website/docs/topics/code-execution/kubernetes-pod-commandline-code-executor.ipynb

LICENSE
- Added Apache License 2.0 to the project because it includes the Kubernetes Python SDK, which is licensed under Apache License 2.0.

## Related issue number

#2967 

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

ekzhu commented:
@questcollector hey thanks for the PR! We will get back to you in a few days.

rysweet commented:
This PR is against AutoGen 0.2. AutoGen 0.2 has been moved to the 0.2 branch. Please rebase your PR on the 0.2 branch or update it to work with the new AutoGen 0.4 that is now in main.

questcollector commented:
> This PR is against AutoGen 0.2. AutoGen 0.2 has been moved to the 0.2 branch. Please rebase your PR on the 0.2 branch or update it to work with the new AutoGen 0.4 that is now in main.

I'll align it with 0.4 version.👌

questcollector commented:
> This PR is against AutoGen 0.2. AutoGen 0.2 has been moved to the 0.2 branch. Please rebase your PR on the 0.2 branch or update it to work with the new AutoGen 0.4 that is now in main.

I rebased it on 0.2 branch. I'll update kubernetes pod commandline code executor to align autogen 0.4 as new PR👌

rysweet commented:
@questcollector - thanks! well done. I think there are just a couple of pre-commit checks to fix to green the CI

jackgerrits commented:
we're happy to take this for 0.2, for 0.4 how do the guidelines seem to you (https://microsoft.github.io/autogen/dev/user-guide/extensions-user-guide/index.html#creating-your-own-community-extension) @questcollector ?

questcollector commented:
> @questcollector - thanks! well done. I think there are just a couple of pre-commit checks to fix to green the CI

oops 😅 I applied pre-commit. Thank you.
@jackgerrits 👌Thank you😊. I will make pod commandline executor by creating community extension.

jackgerrits commented:
Looks like there are still CI issues here that need to be fixed before this can be merged.

questcollector commented:
> Looks like there are still CI issues here that need to be fixed before this can be merged. (I reverted a seemingly unrelated change to gitignore)

Oh sorry for extra works, I fixed test code.
Code block importing kubernetes package moved into try block.
I think it will prevent issues about build actions.
Thank you for reviewing this PR
